src/test/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSRIntegrationTest.java

```diff
@@ -90,18 +96,82 @@ public void testApplyBQSR(ABQSRTest params) throws IOException {
         args.add(new File(params.bam).getAbsolutePath());
         args.add("--" + StandardArgumentDefinitions.BQSR_TABLE_LONG_NAME);
         args.add(new File(resourceDir + "HiSeq.20mb.1RG.table.gz").getAbsolutePath());
-        args.add("-O"); args.add(outFile.getAbsolutePath());
+        args.add("-O");
+        args.add(outFile.getAbsolutePath());
         if (params.reference != null) {
             refFile = new File(params.reference);
-            args.add("-R"); args.add(refFile.getAbsolutePath());
+            args.add("-R");
+            args.add(refFile.getAbsolutePath());
+            if (params.args != null) {
+                Stream.of(params.args).forEach(arg -> args.add(arg));
+            }
+
+            runCommandLine(args);
+
+            SamAssertionUtils.assertSamsEqual(outFile, new File(params.expectedFile), refFile);
         }
-        if (params.args != null) {
-            Stream.of(params.args).forEach(arg -> args.add(arg));
+    }
+
+    @Test(dataProvider = "ApplyBQSRTest")
+    public void testApplyBQSRPath(ABQSRTest params) throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path outPath = jimfs.getPath("applyBQSRTest"+params.outputExtension);
+
+            final ArrayList<String> args = new ArrayList<>();
+            Path refPath = null;
+
+            args.add("-I");
+            args.add(new File(params.bam).getAbsolutePath());
+            args.add("--" + StandardArgumentDefinitions.BQSR_TABLE_LONG_NAME);
+            args.add(new File(resourceDir + "HiSeq.20mb.1RG.table.gz").getAbsolutePath());
+            args.add("-O"); args.add(outPath.toUri().toString());
+            if (params.reference != null) {
+                File refFile = new File(params.reference);
+                args.add("-R"); args.add(refFile.getAbsolutePath());
+                refPath = refFile.toPath();
+            }
+            if (params.args != null) {
+                Stream.of(params.args).forEach(arg -> args.add(arg));
+            }
+
+            runCommandLine(args);
+
+            SamAssertionUtils.assertSamsEqual(outPath, new File(params.expectedFile).toPath(), refPath);
         }
+    }
 
-        runCommandLine(args);
+    @Test(dataProvider = "ApplyBQSRTest", groups={"cloud", "bucket"})
```

Comment:
This test should be just in the `bucket` group, not the `cloud` group, since execution is local

Reply:
Good catch, thank you! Fixed!